addons/api/addon/handlers/client-daemon-handler.js

```diff
@@ -21,7 +22,7 @@ const ClientDaemonHandler = {
     switch (context.request.op) {
       case 'query': {
         const { store, data } = context.request;
-        const { type, query } = data;
+        const { type, query, options: { pushToStore = true } = {} } = data;
```

Comment:
This is just an extra escape hatch I added for scenarios where we want to retrieve data without pushing to the store.